### PR TITLE
Fix #4887 -  CSharp workflow - macos-10.15 was removed

### LINKED ISSUE #4887: CSharp workflow is failing due to macos-10.15 being removed (Nuget)
Issue overview
--------------

Nuget failed for 3.6.0: https://github.com/NREL/OpenStudio/actions/runs/4927921801

image (6)

## Possible Solution

Update to macos-11

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #4887

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: .github/workflows/buildCSharp.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: "Checkout develop Branch"
         uses: actions/checkout@v3
         with:
-          ref: develop
+          ref: master
```
Comment (jmarrec) [inline .github/workflows/buildCSharp.yml:35]: This is temporary, trying to generate the nuget for 3.6.0 again at https://github.com/NREL/OpenStudio/actions/runs/4989940843

Also, I don't think we generally always want to build develop, do we @tijcolem ? We should omit the `ref` and let Github figure out what to build: a push to master would build master, a push of a tag would build the tag, a PR to master would build the resulting commit of the PR merged into master. Doesn't that make more sense?

File: .github/workflows/buildCSharp.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
         - name: Ubuntu
           os: ubuntu-20.04
         - name: macOS
-          os: macos-10.15
+          os: macos-11
```
Comment (jmarrec) [inline .github/workflows/buildCSharp.yml:25]: Actual change is here, the rest is just linting the workflow file.

File: .github/workflows/buildCSharp.yml (hunk 1 above)
Comment (jmarrec) [reply .github/workflows/buildCSharp.yml:35]: https://github.com/NREL/OpenStudio/actions/runs/4989940843 worked. @tijcolem

Comment (jmarrec) [reply .github/workflows/buildCSharp.yml]: @MingboPeng I think you manually reupload these at the platform specific nuget endpoint?
eg https://www.nuget.org/packages/NREL.OpenStudio.macOS-x64, https://www.nuget.org/packages/NREL.OpenStudio.linux-x64, https://www.nuget.org/packages/NREL.OpenStudio.win

Comment (MingboPeng) [reply .github/workflows/buildCSharp.yml]: Hi @jmarrec yes, I manually uploaded those packages.

Comment (jmarrec) [reply .github/workflows/buildCSharp.yml]: @MingboPeng If you want to do that again, https://github.com/NREL/OpenStudio/actions/runs/4989940843 if where you could get them.

**Pull request merged**